emscripten.py

```diff
@@ -39,6 +40,20 @@
   logging.info('logging stderr in js compiler phase into %s' % STDERR_FILE)
   STDERR_FILE = open(STDERR_FILE, 'w')
 
+# Global state that tracks the set of used (minified) function names in
+# JS symbols imported to asm.js module.
+minified_js_names = MinifiedJsNameGenerator()
+
+
+# Converts list of imports ['foo', 'bar', ...] to a dictionary of
+# name mappings in form { 'minified': 'unminified', ... }
+def define_asmjs_import_names(imports, minify_names):
+  minified = {}
+  for name in imports:
+    import_name = minified_js_names.generate() if minify_names else name
+    minified[import_name] = name
+  return minified
```

Comment:
Maybe move this down to where its used?  Could even be an inner function within `function_tables_and_exports`?

Also we can drop the second argument and just use shared.Settings.MINIFY_ASMJS_IMPORT_NAMES directly?

Also what do you think of:

```
if shared.Settings.MINIFY_ASMJS_IMPORT_NAMES:
    return {minified_js_names.generate(): i for i in imports}
else:
    return {i: i for i in imports}
```